subsys/bluetooth/services/dis.c

```diff
@@ -128,12 +151,20 @@ static uint8_t dis_ieee_rcdl[CONFIG_BT_DIS_STR_MAX + 1] = CONFIG_BT_DIS_IEEE_RCD
 
 #endif /* CONFIG_BT_DIS_SETTINGS */
 
+#define BT_DIS_READ_STR_USED \
+	(CONFIG_BT_DIS_MODEL_NUMBER || CONFIG_BT_DIS_MODEL_DEPRECATED_USED || \
+	 CONFIG_BT_DIS_MANUF_NAME || CONFIG_BT_DIS_MANUF_DEPRECATED_USED || \
+	 CONFIG_BT_DIS_SERIAL_NUMBER || CONFIG_BT_DIS_FW_REV || CONFIG_BT_DIS_HW_REV || \
+	 CONFIG_BT_DIS_SW_REV || CONFIG_BT_DIS_IEEE_RCDL)
+
+#if BT_DIS_READ_STR_USED
```

Comment:
Can't we rely on linker to remove the function if unused? We could add `ARG_UNUSED(read_str);` to prevent build warnings if needed

Reply:
Discussed offline. We can leave it as is (finding proper place for `ARG_UNUSED(read_str);` is problematic)